From 2ff966e140156243a82e452de04b839750326e89 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Wed, 3 Oct 2018 14:28:57 -0700
Subject: [PATCH 6/8] DO-NOT-UPSTREAM: Hide -Wempty-body behind W=1

This doesn't happen often but the few times that it has, it's just
noise. I believe the intention of the warning is to avoid situations
like this:

if (condition);
        statement;

where the user really intended

if (condition)
        statement;

GCC's -Wmisleading-indention already catches these issues though so this
warning just becomes noise.

Link: https://github.com/ClangBuiltLinux/linux/issues/42
Link: https://github.com/ClangBuiltLinux/linux/issues/66
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 Makefile                   | 1 +
 scripts/Makefile.extrawarn | 1 +
 2 files changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index cc85f2d50c82..5f9c95493df2 100644
--- a/Makefile
+++ b/Makefile
@@ -703,6 +703,7 @@ KBUILD_CFLAGS += $(call cc-disable-warning, address-of-packed-member)
 # Quiet clang warning: comparison of unsigned expression < 0 is always false
 KBUILD_CFLAGS += $(call cc-disable-warning, tautological-compare)
 KBUILD_CFLAGS += $(call cc-disable-warning, duplicate-decl-specifier)
+KBUILD_CFLAGS += $(call cc-disable-warning, empty-body)
 # CLANG uses a _MergedGlobals as optimization, but this breaks modpost, as the
 # source of a reference will be _MergedGlobals and not on of the whitelisted names.
 # See modpost pattern 2
diff --git a/scripts/Makefile.extrawarn b/scripts/Makefile.extrawarn
index 8d5357053f86..9bd19cec7b56 100644
--- a/scripts/Makefile.extrawarn
+++ b/scripts/Makefile.extrawarn
@@ -31,6 +31,7 @@ warning-1 += $(call cc-option, -Wunused-const-variable)
 warning-1 += $(call cc-option, -Wpacked-not-aligned)
 warning-1 += $(call cc-disable-warning, missing-field-initializers)
 warning-1 += $(call cc-disable-warning, sign-compare)
+warning-1 += $(call cc-option, -Wempty-body)
 
 warning-2 := -Waggregate-return
 warning-2 += -Wcast-align
-- 
2.19.0

